src/lib/assetBridger/erc20Bridger.ts

```diff
@@ -640,27 +697,75 @@ export class Erc20Bridger extends AssetBridger<
     return L1TransactionReceipt.monkeyPatchContractCallWait(tx)
   }
 
+  /**
+   * Get the arguments for calling the token withdrawal function
+   * @param params
+   * @returns
+   */
+  public async getWithdrawalParams(
+    params: Erc20WithdrawParams
+  ): Promise<L2ToL1TransactionRequest> {
+    if (!SignerProviderUtils.signerHasProvider(params.l2Signer)) {
+      throw new MissingProviderArbSdkError('l2Signer')
+    }
+    await this.checkL2Network(params.l2Signer)
+
+    const to = params.destinationAddress || (await params.l2Signer.getAddress())
+
+    const routerInterface = L2GatewayRouter__factory.createInterface()
+    const functionData =
+      // we need to do this since typechain doesnt seem to correctly create
+      // encodeFunctionData for functions with overrides
```

Comment:
latest typechain version also has this bug? if so, would be cool if we opened an issue in their repo

Reply:
interesting it seems to be present in the export
https://unpkg.com/browse/@arbitrum/sdk@2.0.13/dist/lib/abi/L2GatewayRouter.d.ts
```ts
encodeFunctionData(functionFragment: "outboundTransfer", values: [string, string, BigNumberish, BytesLike]): string;
```

Reply:
so the underlying contract has 2 functions on it both called "outboundTransfer" but with different args. So they need to be called with something like "outboundTransfer(address,address,uint256,bytes)" so that the underlying contract finds the correct one. The typechain interface only "outboundTransfer" - which isnt actually a func on the underlying ethers contract object.

Yeh, we need to create a ticket, or check if this is the case after upgrading.

Reply:
Yeh, looks like this is still an issue in the latest typechain

I'll create an issue

Reply:
https://github.com/dethcrypto/TypeChain/issues/752